source/handlers/client.py

```diff
@@ -33,15 +30,30 @@ def __init__(self, sock, clients, ip, port):
     def run(self):
         while self._running:
             try:
-                request = self._recieve_request()
-                self._socket.send(json.dumps(request))
+                request = Request.deserialize(self._recieve_request())
+                self.handle_request(request.code, request.args)
             except ValueError:
                 self._send_bad_request()
             except socket.timeout:
                 continue
             except socket.error:
                 return self.close()
 
+
+    def handle_request(self, code, args):
+        try:
+            controller_function = get_controller_func(code)
+            
+            if controller_function:
+                response = controller_function(args)
+                self._send_message(response.serialize())
+            else:
```

Comment:
Not needed. If a controller_func is not found an error is thrown. you can remove the if/else clause.